### PR TITLE
Fix for routing_table tests, Implement public_id name

### PR DESCRIPTION
**Pull request by dirvine**
[Review on Reviewable](https://reviewable.io/reviews/maidsafe/routing/486)

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
r? @benjaminbollen

(maidsafe_highfive has picked a reviewer for you, use r? to override)

File: src/id.rs [hunk 1]
```diff
@@ -53,10 +56,11 @@ impl Id {
 
     pub fn with_keys(sign_keys: (crypto::sign::PublicKey, crypto::sign::SecretKey),
                      encrypt_keys: (crypto::box_::PublicKey, crypto::box_::SecretKey))-> Id {
+        let name = NameType::new(crypto::hash::sha512::hash(&sign_keys.0[..]).0);
```
Comment (inetic) [inline src/id.rs]: Could we use this function?
https://github.com/maidsafe/routing/blob/master/src/utils.rs#L42
Or does that one have different semantic?

Comment (dirvine) [inline src/id.rs]: Yes I think we will end up being more explicit here and using only the publicKey for non xor nodes.

**Pull request merged**